src/shared/components/data-table/components/main/components/table-body/TableBody.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useContext } from 'react';
+
+import { defaultTableState } from 'src/shared/components/data-table/dataTableReducer';
+import { TableContext } from 'src/shared/components/data-table/DataTable';
+import TableRow from '../table-row/TableRow';
+
+import { SortingDirection } from 'src/shared/components/data-table/dataTableTypes';
+
+const TableBody = () => {
+  const {
+    data,
+    currentPageNumber,
+    rowsPerPage,
+    columns,
+    sortedColumn,
+    uniqueColumnId
+  } = useContext(TableContext) || {
+    currentPageNumber: defaultTableState.currentPageNumber,
+    rowsPerPage: defaultTableState.rowsPerPage
+  };
+
+  if (!(data && columns)) {
+    return null;
+  }
+  const totalRows = data.length;
+  const rowIndexLowerBound = (currentPageNumber - 1) * rowsPerPage;
+  const rowIndexUpperBound =
+    (currentPageNumber - 1) * rowsPerPage + rowsPerPage;
```

Comment:
Won't this give more context?

`const rowIndexUpperBound = rowIndexLowerBound + rowsPerPage;`

Reply:
That's done